### ISSUE TITLE
Zappi status is always "Boosting"

### ISSUE THREAD

**Issue by JonSilver** [CJNE/ha-myenergi#391]
## Version of the custom_component
0.0.23

## Describe the bug

Ever since installing and configuring the integration, my the "status" sensor for my Zappi has been displayed as "Boosting", even when the charge is stopped, or it's waiting for surplus or whatever.

**Comment by reedy** [comment]
![Screenshot 2023-09-13 at 09 43 03](https://github.com/CJNE/ha-myenergi/assets/67615/a0cb56f2-b9c5-40e0-b4fc-3c746a1e05ae)

Seems to be working fine for me (but my Zappi has only been in for a week or so).

Is the sensor reporting (correctly or otherwise) that it has updated/changed recently?

**Comment by JonSilver** [comment]
Yes, it's reported to have changed at the same time as the others but the actual value didn't change.

![image](https://github.com/CJNE/ha-myenergi/assets/8831106/6b9df345-02ad-4b11-ae73-e17791409333)
![image](https://github.com/CJNE/ha-myenergi/assets/8831106/52440835-92ef-47d4-b4e7-40d346cc880a)

**Comment by G6EJD** [comment]
I’ve just checked mine, it’s working perfectly, it’s much more likely that your Zappi is losing contact with the Master or Hub.

**Comment by plord12** [comment]
Many thanks for checking.  Yes, we are suspecting a fault on the zappi.

**Comment by JonSilver** [comment]
This is the 2nd Zappi. The first was replaced when its screen spontaneously smashed due to an installation error. Both have exhibited this behaviour on HA, but were quite certain themselves that they were stopped and not "boosting". Is this merely a misinterpretation of what comes out of the API?

**Comment by steve-hawkins** [comment]
I can confirm this is the case for me as well since 0.0.23, but also still present in 0.0.24

I have a zappi v2.1 with internal hub

when the EV is plugged in, the device status will change from boosting to completed after a charging session:

![image](https://github.com/CJNE/ha-myenergi/assets/7125740/a7f5186d-d378-448f-8674-bb2a4f1927f4)

P.S.

any idea why there is a typo on 'myenergi zappi **Plus** Status'?

**Comment by reedy** [comment]
I have seen it once or twice on mine now...

![Screenshot 2023-12-27 at 15 01 25](https://github.com/CJNE/ha-myenergi/assets/67615/af7796f2-1d00-4db0-8bf2-411c46991a65)

**Comment by reedy** [comment]
Mine is currently boosting (apparently) with EV disconnected...

![image](https://github.com/CJNE/ha-myenergi/assets/67615/8a7a86d9-4446-488a-93d3-761dd2752d69)

**Comment by steve-hawkins** [comment]
has anyone been able to make any progress on this one?

**Comment by G6EJD** [comment]
Mine is working without issue, what entity name are you using?
'myenergi EV Charger Status'

There is also a plug status

**Comment by steve-hawkins** [comment]
> Mine is working without issue, what entity name are you using? 'myenergi EV Charger Status'
> 
> There is also a plug status

The same situation as I posted in Dec:

https://github.com/CJNE/ha-myenergi/issues/391#issuecomment-1870174873

could this be due to the API changes mentioned here #217 on the zappi v2.1 with the internal hub?

**Comment by LeiChat** [comment]
Not seeing the issue with my zappi 2.1 (with vHub) running firmware 3562S5.044
![image](https://github.com/CJNE/ha-myenergi/assets/23559551/98eaf9dc-19cb-46fd-9fab-555809bb26d3)
I am using v.0.0.27 of the myenergi HA integration.

**Comment by G6EJD** [comment]
It’s still worth checking your entities with a search for ‘status’ there should be just two.

**Comment by MalcolmSpencer** [comment]
I saw this and concluded that it matched the times on the scheduled boost settings even though boost was set to manual and the Zappi in stop mode. I don't yet own an EV and have never plugged one into my Zappi so not sure how it thinks its boosting. Sounds like a Zappi or Myenergi bug.

**Comment by G6EJD** [comment]
This issue needs to be closed by the instigator.

**Comment by AJediIAm** [comment]
> Use 👍 on one of the options below to vote.

**Option 2: override status when unplugged**
When Charge mode is set to "Fast" and Plug status is "EV Disconnected", override the status to "Paused"

**Comment by G6EJD** [comment]
Myenergi has chosen to set the Zappi status to Boosting when in Fast mode, or any state when the EV is charging. Only when the plug is removed does the status of boosting get removed.

It's proposed to close this issue after 10-days if there is no update from the originator.
